### PR TITLE
v22.1.14 release notes

### PR DESCRIPTION
**Pull request by gemma-shay**
https://cockroachlabs.atlassian.net/browse/DOC-6400

### CODE REVIEW

File: _includes/releases/v22.1/v22.1.14.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
+## v22.1.14
+
+Release Date: February 6, 2023
+
+{% include releases/release-downloads-docker-image.md release=include.release %}
+
+<h3 id="v22-1-14-sql-language-changes">SQL language changes</h3>
+
+- [`COPY`](../v22.1/copy-from.html) now logs an error during the insert phase on the [`SQL_EXEC`](../v22.1/logging.html#sql_exec) logging channel. [#95175][#95175]
+- If `copy_from_retries_enabled` is set, [`COPY`](../v22.1/copy-from.html) is now able to retry certain safe circumstances. For example, when `copy_from_atomic_enabled` is false, there is no transaction running `COPY` and the error returned is retriable. [#95505][#95505]
```
Comment (rmloveland) [inline _includes/releases/v22.1/v22.1.14.md:10]: Is `copy_from_retries_enabled` a session setting?  If so I suggest editing to something like

> the `copy_from_retries_enabled` [session setting](https://www.cockroachlabs.com/docs/v22.1/set-vars)

Later on in the sentence I see a missing article "is able to retry _in_ certain safe circumstances"  (could also be 'under' I think)

File: _includes/releases/v22.1/v22.1.14.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+- `kv.bulkio.write_metadata_sst.enabled` now defaults to false. This change does not affect [`BACKUP`](../v22.1/backup.html) or [`RESTORE`](../v22.1/restore.html). [#96017][#96017]
```
Comment (rmloveland) [inline _includes/releases/v22.1/v22.1.14.md:11]: Suggest editing to 

> The `kv....` [cluster setting](https://www.cockroachlabs.com/docs/v22.1/cluster-settings)

File: _includes/releases/v22.1/v22.1.14.md [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+<h3 id="v22-1-14-db-console-changes">DB Console changes</h3>
+
+- Removed the [**Reset SQL stats**](../v22.1/ui-statements-page.html) and [**Reset index stats**](../v22.1/ui-databases-page.html#index-details) buttons from the DB Console for non-admin users. [#95325][#95325]
+- [Graphs](../v22.1/ui-overview-dashboard.html) can now be clicked on to toggle legend "stickiness" and make the points stop following the mouse. This makes it easier to read dense graphs with many series plotted together. [#94786][#94786]
+
+<h3 id="v22-1-14-bug-fixes">Bug fixes</h3>
+
+- Fixed a bug where, in a cluster with nodes running both [v22.2](v22.2.html) and v22.1, range replica changes could sometimes fail on v22.1 leaseholders with the error `change replicas of r47 failed: descriptor changed: [expected] != [actual]`, without any apparent differences between the listed descriptors. Continuing to upgrade all nodes to v22.2 or rolling all nodes back to v22.1 would resolve this issue. [#94841][#94841]
```
Comment (rmloveland) [inline _includes/releases/v22.1/v22.1.14.md:20]: 'range replica' could be a link to https://www.cockroachlabs.com/docs/stable/ui-replication-dashboard.html#review-of-cockroachdb-terminology

File: _includes/releases/v22.1/v22.1.14.md [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+- It is now possible to run [`cockroach version`](../v22.2/cockroach-version.html) and [`cockroach start`](../v22.2/cockroach-start.html) (and possibly other sub-commands) when the user running the command does not have permission to access the current working directory. [#94926][#94926]
+- Fixed a bug where [`CLOSE ALL`](../v22.1/sql-grammar.html#close_cursor_stmt) would not respect the `ALL` flag and would instead attempt to close a cursor with no name. [#95440][#95440]
+- Fixed a crash that could happen when formatting a tuple with an unknown type. [#95422][#95422]
+- Fixes a bug where a [database restore](../v22.1/restore.html) would not grant `CREATE` and `USAGE` on the public schema to the public role. [#95537][#95537]
```
Comment (rmloveland) [inline _includes/releases/v22.1/v22.1.14.md:24]: "Fixes" could be changed to "Fixed" to match other items

mention of 'grant' could be a link to https://www.cockroachlabs.com/docs/v22.1/grant.html 

Similarly 'role' could link to https://www.cockroachlabs.com/docs/v22.1/security-reference/authorization#users-and-roles

File: _includes/releases/v22.1/v22.1.14.md [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+- Fixed a bug where [`pg_get_indexdef`](../v22.1/functions-and-operators.html) didn't show the expression used to define an expression-based index, as well as a bug where the function was incorrectly including columns stored by the index. [#95585][#95585]
```
Comment (rmloveland) [inline _includes/releases/v22.1/v22.1.14.md:25]: 'expression-based index' could link to https://www.cockroachlabs.com/docs/v22.1/partial-indexes

File: _includes/releases/v22.1/v22.1.14.md [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+- Fixed a bug where a DNS lookup was performed during gossip remote forwarding while holding the gossip mutex. This could cause processing stalls if the DNS server was slow to respond. [#95441][#95441]
+- Fixed a bug where [`RESTORE SYSTEM USERS`](../v22.1/restore.html#restoring-users-from-system-users-backup) would fail to restore role options. [#95295][#95295]
```
Comment (rmloveland) [inline _includes/releases/v22.1/v22.1.14.md:27]: 'role options' could be a link to https://www.cockroachlabs.com/docs/stable/security-reference/authorization.html#role-options

File: _includes/releases/v22.1/v22.1.14.md [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+- Reduce register session, deregister session, and session cancel query contention. [#95654][#95654]
+- Fixed a bug where a backup of keys with many revisions would fail with `pebble: keys must be added in order`. [#95446][#95446]
+- Fixed the `array_to_string` [built-in function](../v22.1/functions-and-operators.html) so that nested arrays are traversed without printing `ARRAY` at each nesting level. [#95844][#95844]
```
Comment (rmloveland) [inline _includes/releases/v22.1/v22.1.14.md:30]: the mention of 'arrays' could link to https://www.cockroachlabs.com/docs/v22.1/array

File: _includes/releases/v22.1/v22.1.14.md [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+- Fixed a crash that could happen when formatting a tuple with an unknown type. [#95422][#95422]
+- Fixes a bug where a [database restore](../v22.1/restore.html) would not grant `CREATE` and `USAGE` on the public schema to the public role. [#95537][#95537]
+- Fixed a bug where [`pg_get_indexdef`](../v22.1/functions-and-operators.html) didn't show the expression used to define an expression-based index, as well as a bug where the function was incorrectly including columns stored by the index. [#95585][#95585]
+- Fixed a bug where a DNS lookup was performed during gossip remote forwarding while holding the gossip mutex. This could cause processing stalls if the DNS server was slow to respond. [#95441][#95441]
+- Fixed a bug where [`RESTORE SYSTEM USERS`](../v22.1/restore.html#restoring-users-from-system-users-backup) would fail to restore role options. [#95295][#95295]
+- Reduce register session, deregister session, and session cancel query contention. [#95654][#95654]
+- Fixed a bug where a backup of keys with many revisions would fail with `pebble: keys must be added in order`. [#95446][#95446]
```
Comment (rmloveland) [inline _includes/releases/v22.1/v22.1.14.md:29]: Mention of 'backup' could link to https://www.cockroachlabs.com/docs/v22.1/backup

File: _includes/releases/v22.1/v22.1.14.md [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+- Fixed a bug where [`CLOSE ALL`](../v22.1/sql-grammar.html#close_cursor_stmt) would not respect the `ALL` flag and would instead attempt to close a cursor with no name. [#95440][#95440]
+- Fixed a crash that could happen when formatting a tuple with an unknown type. [#95422][#95422]
+- Fixes a bug where a [database restore](../v22.1/restore.html) would not grant `CREATE` and `USAGE` on the public schema to the public role. [#95537][#95537]
+- Fixed a bug where [`pg_get_indexdef`](../v22.1/functions-and-operators.html) didn't show the expression used to define an expression-based index, as well as a bug where the function was incorrectly including columns stored by the index. [#95585][#95585]
+- Fixed a bug where a DNS lookup was performed during gossip remote forwarding while holding the gossip mutex. This could cause processing stalls if the DNS server was slow to respond. [#95441][#95441]
+- Fixed a bug where [`RESTORE SYSTEM USERS`](../v22.1/restore.html#restoring-users-from-system-users-backup) would fail to restore role options. [#95295][#95295]
+- Reduce register session, deregister session, and session cancel query contention. [#95654][#95654]
```
Comment (rmloveland) [inline _includes/releases/v22.1/v22.1.14.md:28]: Suggest rephrasing to 'Reduce contention between queries to register, deregister, and cancel sessions'

And make 'sessions' a link to https://www.cockroachlabs.com/docs/v22.1/show-sessions

File: _includes/releases/v22.1/v22.1.14.md [hunk 10]
```diff
@@ -0,0 +1,62 @@
...
+- Fixed a bug where a backup of keys with many revisions would fail with `pebble: keys must be added in order`. [#95446][#95446]
+- Fixed the `array_to_string` [built-in function](../v22.1/functions-and-operators.html) so that nested arrays are traversed without printing `ARRAY` at each nesting level. [#95844][#95844]
+- Fixed a bug that caused ranges to remain without a leaseholder in cases of asymmetric network partitions. [#95237][#95237]
```
Comment (rmloveland) [inline _includes/releases/v22.1/v22.1.14.md:31]: Suggest 'ranges' be a link to https://www.cockroachlabs.com/docs/v22.1/architecture/overview#architecture-range

Suggest 'network partitions' be a link to https://www.cockroachlabs.com/docs/v22.1/cluster-setup-troubleshooting.html#network-partition

File: _includes/releases/v22.1/v22.1.14.md [hunk 11]
```diff
@@ -0,0 +1,62 @@
...
+- Fixed a bug where [`COPY`](../v22.1/copy-from.html) into a column with collated strings would result in an error similar to `internal error: unknown type collatedstring`. [#96039][#96039]
```
Comment (rmloveland) [inline _includes/releases/v22.1/v22.1.14.md:32]: Suggest making 'collated strings' a link to https://www.cockroachlabs.com/docs/v22.1/collate

File: _includes/releases/v22.1/v22.1.14.md [hunk 12]
```diff
@@ -0,0 +1,62 @@
...
+- Fixed a bug where spurious transaction restarts could occur when validating a [`FOREIGN KEY`](../v22.1/foreign-key.html) in the same transaction where the referenced table is modified. If the transaction was running at `PRIORITY HIGH`, deadlocks could occur. [#96124][#96124]
```
Comment (rmloveland) [inline _includes/releases/v22.1/v22.1.14.md:33]: suggest making 'transaction restarts' a link to https://www.cockroachlabs.com/docs/stable/common-errors.html#restart-transaction

suggest making `PRIORITY HIGH` a link to https://www.cockroachlabs.com/docs/v22.1/transactions#transaction-priorities

**Review by rmloveland — APPROVED** [review]
LGTM modulo a couple of small rephrasing and wanting more links :-)

File: _includes/releases/v22.1/v22.1.14.md (hunk 1 above)
Comment (gemma-shay) [reply _includes/releases/v22.1/v22.1.14.md:10]: i think it probably is but it's not documented anywhere that i can find... same with the kv cluster setting below this. should i just link to the session/cluster settings pages anyway?

Comment (gemma-shay) [reply _includes/releases/v22.1/v22.1.14.md]: going to merge as-is, can follow up if necessary

**Pull request merged**